### ISSUE TITLE
Table editing mode doesn't interpret existing unaligned columns properly

### ISSUE THREAD

**Issue by smithfred** [retext-project/retext#341]
If ReText is given an existing table with unaligned columns, it tries to "keep alignment" by naïvely looking vertically for table column delimiters according to the *character position* rather than the actual table column number.

So with the below, adding to the "g" cell starts extending the width of "some content" (since it "finds" its RHS delimiter above first) rather than the "a" cell.

```
| some content | some other content | a | b | c | d |
| e | f | g | h | i | j |
```

This makes it pretty much impossible to use table editing mode on an existing unaligned table without first painstakingly aligning everything by hand.

**Comment by mitya57** [comment]
@Griffon26 What do you think about this?

**Comment by Griffon26** [comment]
I agree that it would be better for Markdown tables to align based on column numbers, because those tables don't support colspan. We'll have to make it conditional on the markup type though to not break the behaviour for Restructured Text, which *does* support colspan.

**Comment by Griffon26** [comment]
I've started working on a fix and pushed it to the unalignedcolumns branch. I haven't yet fixed it for RestructuredText and I also still have to fix the test cases.

**Comment by Griffon26** [comment]
Fixed in commit 7d836cd. Thanks for the good idea, @smithfred!

**Comment by smithfred** [comment]
Thanks for the fast fix! 😄 👍
